ement-room.el

```diff
@@ -2787,135 +2800,118 @@ ROOM defaults to the value of `ement-room'."
 
-(defun ement-room--cleanse-url-preview (data link)
-  "Ensure a url preview DATA always has a title and associated LINK."
-  (unless (alist-get 'og:url data) (push (cons 'og:url link) data))
-  (unless (alist-get 'og:title data)
-    (push (cons 'og:title (replace-regexp-in-string "\\<https?://\\|/.*$\\|/$" "" link)) data))
-  data)
-
-(defun ement-room--insert-preview (msg related num data)
-  "Insert link preview for DATA into MSG from node RELATED if it isn't already present.
+(defun ement-room--insert-preview (message related num data)
+  "Insert link preview for DATA into MESSAGE from node RELATED if it isn't already present.
 NUM is the index for this link."
-  (cond ((alist-get 'errcode data) (progn
-                                     (message "Ement: You are being rate limited. Waiting 10s before trying again.")
-                                     (setq-local ement-room--preview-stop t)
-                                     (let ((saved-room (current-buffer)))
-                                       (run-with-timer 10 nil
-                                                       (lambda ()
-                                                         (with-current-buffer saved-room
-                                                           (setq-local ement-room--preview-stop nil)))))))
+  (cond ((alist-get 'errcode data) (ement-debug "Rate limit in url preview"))
         (data (let ((related-node (ement-room--ewoc-node-before ement-ewoc
-                                                                msg
+                                                                message
                                                                 (lambda (a b)
                                                                   (string= (ement-event-id a)
                                                                            (ement-event-id b)))
                                                                 :pred #'ement-event-p))
                     (key (intern (concat "preview-" (number-to-string num)))))
-                (unless (alist-get key (ement-event-local related))
-                  (cl-pushnew (cons key data) (ement-event-local related)))
-                (when (> num (or (map-elt (ement-event-local related) 'url-preview-n)
-                                 most-negative-fixnum))
-                  (setf (alist-get 'url-preview-n (ement-event-local related)) num))
+                (when (not (alist-get 'url-previews (ement-event-local related)))
+                  (cl-pushnew (cons 'url-previews nil) (ement-event-local related)))
+                (unless (alist-get key (map-elt (ement-event-local related) 'url-previews))
+                  (cl-pushnew (cons key data) (map-elt (ement-event-local related) 'url-previews)))
                 (ewoc-invalidate ement-ewoc related-node)))))
 
 (defun ement-room--format-url-preview (data)
-  "Format a pretty-printed preview for DATA."
+  "Return a pretty-printed preview for DATA."
```

Comment:
Please mention what `DATA` is on a second line in the docstring (i.e. in this case I think it's the JSON from the parsed HTTP response).